server/src/main/java/org/elasticsearch/action/IndicesRequest.java

```diff
@@ -48,9 +49,25 @@ interface Replaceable extends IndicesRequest {
          */
         IndicesRequest indices(String... indices);
 
+        /**
+         * Record the results of index resolution. See {@link ResolvedIndexExpressions} for details.
+         * Note: this method does not replace {@link #indices(String...)}. {@link #indices(String...)} must still be called to update
```

Comment:
Nothing to do for this PR. But could you please remind me what the long term vision is for this new method and the existing `indices(String ...indices)` method? Did we align that the new one could eventually replace the existing one and as an interim the default implementation could call something like `indices(expressions.getAllLocalIndices())`? Or maybe something else? My memory is hazy on this one.

Reply:
My current plan for the interim is to maintain the methods separately and keep them independent. However, I'll still play around with the option of making a call like `indices(expressions.getAllLocalIndices())` under the hood in the next PR where I plug this into the indices resolver. 

Long term, the plan is to consolidate the two methods (replace `indices(...)`). That's a bigger refactor but seems very worthwhile since it'll move us closer to having structured info about index resources as opposed to opaque strings.